diff --git a/arch/arm64/boot/dts/rockchip/Makefile b/arch/arm64/boot/dts/rockchip/Makefile
index 4bf84622db47..b1a8b52d33c9 100644
--- a/arch/arm64/boot/dts/rockchip/Makefile
+++ b/arch/arm64/boot/dts/rockchip/Makefile
@@ -128,6 +128,12 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-box-demo.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-nanopi-r3s.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-bigtreetech-cb2-manta.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-bigtreetech-pi2.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-bigtreetech-pi2-pcie.dtbo
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-bigtreetech-pi2-printer.dtbo
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-bigtreetech-pi2-happ.dtbo
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-bigtreetech-pi2-back_srv.dtbo
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-bigtreetech-pi2-dsi1.dtbo
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-bigtreetech-pi2-sfc.dtbo
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-tinker-board-3.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-tinker-board-3s.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-9tripod-x3568-v4.dtb
@@ -277,6 +281,30 @@ rk3588-rock-5b-plus-radxa-4k-cam-dtbs := rk3588-rock-5b-plus.dtb \
 	rk3588-rock-5b-plus-radxa-cam4k-cam0.dtbo \
 	rk3588-rock-5b-plus-radxa-cam4k-cam1.dtbo
 
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588-tiger-haikou-haikou-video-demo.dtb
 rk3588-tiger-haikou-haikou-video-demo-dtbs := rk3588-tiger-haikou.dtb \
 	rk3588-tiger-haikou-video-demo.dtbo
+
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-bigtreetech-pi2-pcie.dtb
+rk3566-bigtreetech-pi2-pcie-dtbs := rk3566-bigtreetech-pi2.dtb \
+	rk3566-bigtreetech-pi2-pcie.dtbo
+
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-bigtreetech-pi2-sfc.dtb
+rk3566-bigtreetech-pi2-sfc-dtbs := rk3566-bigtreetech-pi2.dtb \
+	rk3566-bigtreetech-pi2-sfc.dtbo
+
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-bigtreetech-pi2-dsi1.dtb
+rk3566-bigtreetech-pi2-dsi1-dtbs := rk3566-bigtreetech-pi2.dtb \
+	rk3566-bigtreetech-pi2-dsi1.dtbo
+
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-bigtreetech-pi2-printer_tst.dtb
+rk3566-bigtreetech-pi2-printer_tst-dtbs := rk3566-bigtreetech-pi2.dtb \
+	rk3566-bigtreetech-pi2-printer.dtbo
+
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-bigtreetech-pi2-happ_tst.dtb
+rk3566-bigtreetech-pi2-happ_tst-dtbs := rk3566-bigtreetech-pi2.dtb \
+	rk3566-bigtreetech-pi2-happ.dtbo
+
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-bigtreetech-pi2-back_srv_tst.dtb
+rk3566-bigtreetech-pi2-back_srv_tst-dtbs := rk3566-bigtreetech-pi2.dtb \
+	rk3566-bigtreetech-pi2-back_srv.dtbo
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-pcie.dtso b/arch/arm64/boot/dts/rockchip/rk3566-bigtreetech-pi2-pcie.dtso
new file mode 100644
index 000000000000..9cb6e8f03685
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3566-bigtreetech-pi2-pcie.dtso
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+
+/dts-v1/;
+/plugin/;
+
+&pcie2x1 {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-pcie.dtso b/arch/arm64/boot/dts/rockchip/rk3566-bigtreetech-pi2-dsi1.dtso
new file mode 100644
index 000000000000..9cb6e8f03685
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3566-bigtreetech-pi2-dsi1.dtso
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+
+/dts-v1/;
+/plugin/;
+
+#include <dt-bindings/soc/rockchip,vop2.h>
+#include <dt-bindings/power/rk3568-power.h>
+
+&hdmi {
+	status = "disabled";
+};
+&hdmi_sound {
+	status = "disabled";
+};
+&i2s0_8ch {
+	status = "disabled";
+};
+&i2s1_8ch {
+	status = "disabled";
+};
+&i2c3 {
+	status = "disabled";
+};
+
+&tft_tp {
+	status = "disabled";
+};
+
+&vp1 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	vp1_out_dsi1: endpoint@ROCKCHIP_VOP2_EP_MIPI1 {
+		reg = <ROCKCHIP_VOP2_EP_MIPI1>;
+		remote-endpoint = <&dsi1_in_vp1>;
+	};
+};
+
+&gpu {
+	status = "okay";
+};
+&vop {
+	status = "okay";
+};
+&vop_mmu {
+	status = "okay";
+};
+&display_subsystem {
+	status = "okay";
+};
+&dsi_dphy1 {
+	status = "okay";
+};
+
+&vp0 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	vp0_out_dsi1: endpoint@ROCKCHIP_VOP2_EP_MIPI1 {
+		reg = <ROCKCHIP_VOP2_EP_MIPI1>;
+		remote-endpoint = <&dsi1_in_vp0>;
+	};
+};
+
+&dsi1 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+	phy-supply = <&vdd_logic>;
+
+	dsi1_panel: panel@0 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "btt-pitft";
+		reg = <0x0>;
+		status = "okay";
+		power-supply = <&vcc3v3_sys>;
+		vddc-supply = <&bl_dsi>;
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				reg = <0>;
+				panel_in_dsi1: endpoint {
+					remote-endpoint = <&dsi1_out_panel>;
+				};
+			};
+		};
+	};
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		dsi1_in: port@0 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0>;
+
+			dsi1_in_vp0: endpoint@0 {
+				remote-endpoint = <&vp0_out_dsi1>;
+			};
+
+			dsi1_in_vp1: endpoint@1 {
+				remote-endpoint = <&vp1_out_dsi1>;
+			};
+		};
+
+		dsi1_out: port@1 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <1>;
+			dsi1_out_panel: endpoint {
+				remote-endpoint = <&panel_in_dsi1>;
+			};
+		};
+	};
+};
+
+&i2c2 {
+	status = "okay";
+	clock-frequency = <100000>;
+	avdd-1v8-supply = <&vcca1v8_image>;
+	avdd-0v9-supply = <&vdda0v9_image>;
+	power-domains = <&power RK3568_PD_VI>;
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	bl_dsi: regulator@45 {
+		compatible = "raspberrypi,7inch-touchscreen-panel-regulator";
+		reg = <0x45>;
+		status = "okay";
+	};
+
+	tp_dsi: touchscreen@38 {
+		compatible = "edt,edt-ft5306";
+		reg = <0x38>;
+		status = "okay";
+
+		vcc-supply = <&vcc3v3_sys>;
+		iovcc-supply = <&vcc_3v3>;
+
+		touchscreen-size-x = <800>;
+		touchscreen-size-y = <480>;
+		touchscreen-inverted-x;
+		touchscreen-inverted-y;
+	};
+};
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-bigtreetech-pi2-printer.dtso b/arch/arm64/boot/dts/rockchip/rk3566-bigtreetech-pi2-sfc.dtso
new file mode 100644
index 000000000000..9cb6e8f03685
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3566-bigtreetech-pi2-sfc.dtso
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+
+/dts-v1/;
+/plugin/;
+
+&sfc {
+	status = "okay";
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	flash@0 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <100000000>;
+		spi-rx-bus-width = <4>;
+		spi-tx-bus-width = <1>;
+	};
+};
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-bigtreetech-pi2-printer.dtso b/arch/arm64/boot/dts/rockchip/rk3566-bigtreetech-pi2-printer.dtso
new file mode 100644
index 000000000000..410763276a6b
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3566-bigtreetech-pi2-printer.dtso
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+
+/dts-v1/;
+/plugin/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/pwm/pwm.h>
+#include <dt-bindings/pinctrl/rockchip.h>
+#include <dt-bindings/power/rk3568-power.h>
+#include <dt-bindings/interrupt-controller/arm-gic.h>
+#include <dt-bindings/interrupt-controller/irq.h>
+#include <dt-bindings/clock/rk3568-cru.h>
+
+&pinctrl {
+    gpio-dht11 {
+        gpio_dht11:gpio-dht11 {
+            rockchip,pins = <0 RK_PA0 RK_FUNC_GPIO &pcfg_pull_none>;
+        };
+    };
+};
+
+&sfc {
+	status = "okay";
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	flash@0 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <100000000>;
+		spi-rx-bus-width = <4>;
+		spi-tx-bus-width = <1>;
+	};
+};
+
+&i2c3 {
+	status = "okay";
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	w1_ms: w1@18 {
+		compatible = "maxim,ds2484";
+		reg = <0x18>;
+		status = "okay";
+	};
+};
+
+&pwm5 {
+    status = "okay";
+};
+
+&{/} {
+	fan_inlet: pwm-fan_inlet {
+		compatible = "pwm-fan";
+		#cooling-cells = <2>;
+		cooling-levels = <0 50 100 150 200 255>;
+		pwms = <&pwm5 0 10000000 PWM_POLARITY_INVERTED>;
+		status = "okay";
+	};
+
+   dht11: dht11 {
+       compatible = "dht11";
+       gpios = <&gpio0 RK_PA0 GPIO_ACTIVE_HIGH>;
+       pinctrl-names = "default";
+       pinctrl-0 = <&gpio_dht11>;
+       status = "okay";
+   };
+};
+&uart9 {
+	pinctrl-0 = <&uart9m1_xfer>;
+	pinctrl-names = "default";
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-bigtreetech-pi2-happ.dtso b/arch/arm64/boot/dts/rockchip/rk3566-bigtreetech-pi2-happ.dtso
new file mode 100644
index 000000000000..410763276a6b
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3566-bigtreetech-pi2-happ.dtso
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+
+/dts-v1/;
+/plugin/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/pwm/pwm.h>
+#include <dt-bindings/pinctrl/rockchip.h>
+#include <dt-bindings/power/rk3568-power.h>
+#include <dt-bindings/interrupt-controller/arm-gic.h>
+#include <dt-bindings/interrupt-controller/irq.h>
+#include <dt-bindings/clock/rk3568-cru.h>
+
+&pwm5 {
+    status = "okay";
+};
+
+&{/} {
+	fan_case: pwm-fan_case {
+		compatible = "pwm-fan";
+		#cooling-cells = <2>;
+		cooling-levels = <0 50 100 150 200 255>;
+		pwms = <&pwm5 0 10000000 PWM_POLARITY_INVERTED>;
+		status = "okay";
+	};
+};
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-bigtreetech-pi2-back_srv.dtso b/arch/arm64/boot/dts/rockchip/rk3566-bigtreetech-pi2-back_srv.dtso
new file mode 100644
index 000000000000..9cb6e8f03685
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3566-bigtreetech-pi2-back_srv.dtso
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+
+/dts-v1/;
+/plugin/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/pwm/pwm.h>
+#include <dt-bindings/pinctrl/rockchip.h>
+#include <dt-bindings/power/rk3568-power.h>
+#include <dt-bindings/interrupt-controller/arm-gic.h>
+#include <dt-bindings/interrupt-controller/irq.h>
+#include <dt-bindings/clock/rk3568-cru.h>
+
+&usb_host0_xhci {
+	status = "disabled";
+};
+
+&usb_host1_xhci {
+	status = "disabled";
+};
+
+/* Disable all GPU-related */
+&vop {
+	status = "disabled";
+};
+&vpu {
+	status = "disabled";
+};
+&vdpu_mmu {
+	status = "disabled";
+};
+&vepu {
+	status = "disabled";
+};
+&vepu_mmu {
+	status = "disabled";
+};
+&vdec {
+	status = "disabled";
+};
+&rga {
+	status = "disabled";
+};
+&vop_mmu {
+	status = "disabled";
+};
+&gpu {
+	status = "disabled";
+};
+&hdmi {
+	status = "disabled";
+};
+&hdmi_sound {
+	status = "disabled";
+};
+&i2c3 {
+	status = "disabled";
+};
+&i2s0_8ch {
+	status = "disabled";
+};
+&i2s1_8ch {
+	status = "disabled";
+};
+&rk_sound {
+	status = "disabled";
+};
+&hdmi_con {
+	status = "disabled";
+};
+&display_subsystem {
+	status = "disabled";
+};
+
+/* Disable BT */
+&uart1 {
+	status = "disabled";
+};
+
+/* Disable unused CAN controller */
+&spi1 {
+	status = "disabled";
+};
+
+&pcie2x1 {
+	status = "disabled";
+};
+
+/* TODO - disable WIFI */
+/*&sdmmc1 {
+	status = "disabled";
+};*/
+
+&sata1 {
+	status = "okay";
+};
+
+&sata2 {
+	status = "okay";
+};
+
+&sfc {
+	status = "okay";
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	flash@0 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <100000000>;
+		spi-rx-bus-width = <4>;
+		spi-tx-bus-width = <1>;
+	};
+};
